### PR TITLE
doc: add crates.io and PyPI notes

### PR DESCRIPTION
**Pull request by JLDLaughlin**
This change is [Reviewable](https://reviewable.io/reviews/materializeinc/materialize/5924)

### CODE REVIEW

File: doc/developer/guide.md [hunk 1]
```diff
@@ -248,6 +248,23 @@ recommendations.
...
+## Publishing code changes
+
+### crates.io
+
+Before publishing internal Rust crates to `crates.io`(https://crates.io/), be sure to
+indicate that `MaterializeInc` is the reponsible maintainer by running the following
+command:
+```shell
+cargo owner --add github:MaterializeInc:crate-owners
+```
```
Comment (quodlibetor) [inline doc/developer/guide.md:260]: should we also remove ourselves as owners, so that everything is owned only by the crate-owners group?

Comment (benesch) [reply doc/developer/guide.md:260]: oh is that possible? If so make sense to add that command here, and then we can hunt everyone down and ask them to run it.

**Review by benesch — APPROVED** [review]
Tysm!

File: doc/developer/guide.md (hunk 1 above)
Comment (quodlibetor) [reply doc/developer/guide.md:260]: Yup, [according to the docs](https://doc.rust-lang.org/cargo/commands/cargo-owner.html) it's 
`cargo owner --remove <login>`

Comment (JLDLaughlin) [reply doc/developer/guide.md:260]: Will add today!

**Pull request merged**